src/dataflow/find_command_result.rs

```diff
@@ -0,0 +1,182 @@
+use crate::config::Config;
+use crate::data::lock::lockfile::Lockfile;
+use crate::data::manifest::Manifest;
+use crate::dataflow::lockfile_packages::LockfileResult;
+use crate::dataflow::manifest_packages::ManifestResult;
+use std::env;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Debug, Fail)]
+pub enum Error {
+    #[fail(
+        display = "Command \"{}\" was not found in the local directory or the global install directory.",
+        _0
+    )]
+    CommandNotFound(String),
+    #[fail(
+        display = "Command \"{}\" was not found in the local directory. There was an error parsing the global lockfile. {}",
+        _0, _1
+    )]
+    CommandNotFoundInLocalDirectoryAndErrorReadingGlobalDirectory(String, String),
+    #[fail(
+        display = "Could not find command \"{}\" because there was a problem with the global directory. {}",
+        _0, _1
+    )]
+    ErrorReadingLocalDirectory(String, String),
+    #[fail(
+        display = "Command \"{}\" exists in lockfile, but corresponding module \"{}\" not found in lockfile.",
+        _0, _1
+    )]
+    CommandFoundButCorrespondingModuleIsMissing(String, String),
+}
+
+pub enum FindCommandResult {
+    CommandNotFound(String),
+    CommandFound(PathBuf, Option<String>, String), // source, args, module name
+    Error(failure::Error),
+}
+
+impl FindCommandResult {
+    fn find_command_in_manifest_and_lockfile<S: AsRef<str>>(
+        command_name: S,
+        manifest: Manifest,
+        lockfile: Lockfile,
+    ) -> Self {
+        match lockfile.get_command(command_name.as_ref()) {
+            Ok(lockfile_command) => {
+                if lockfile_command.package_name == manifest.package.name {
+                    // this is a local module command
+                    let module = manifest.module.as_ref().map(|modules| {
+                        let module = modules.iter().find(|m| m.name == lockfile_command.module);
+                        module
+                    });
+                    match module.unwrap_or(None) {
```

Comment:
⛳️ `module.unwrap_or_default()`
⛳️ `module.ok()` (Result -> Option)

Reply:
⛳️  if `module` isn't used anywhere else, adding this `unwrap` command to the binding makes println debugging a bit easier